src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -114245,5 +114233,792 @@ class Boxed<T>
             var comp = CreateCompilation(source, options: WithNonNullTypesTrue(TestOptions.ReleaseExe));
             CompileAndVerify(comp, expectedOutput: "");
         }
+
+        [Fact, WorkItem(38168, "https://github.com/dotnet/roslyn/issues/38168")]
+        public void MethodTypeInference_ImplementedInterfaces_Indirect_Invariant_ObliviousVersusUnannotated()
+        {
+            var source = @"
+#nullable enable warnings
+static class P
+{
+    static void M(C c)
+    {
+        var x = c.Extension();
+        x.ToString();
+    }
+    public static T Extension<T>(this I<T> source) => throw null!;
+}
+
+class C : I<object>, I2 {}
+
+interface I<T> {}
+
+#nullable enable annotations
+interface I2 : I<object> {}
+";
+            // The introduction of nullability adds a warning
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (13,7): warning CS8645: 'I<object>' is already listed in the interface list on type 'C' with different nullability of reference types.
+                // class C : I<object>, I2 {}
+                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "C").WithArguments("I<object>", "C").WithLocation(13, 7)
+                );
+            var tree = comp.SyntaxTrees.Single();
+            var declaration = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var local = (ILocalSymbol)model.GetDeclaredSymbol(declaration);
+            Assert.Equal("System.Object", local.Type.ToTestDisplayString());
+            Assert.Equal(CodeAnalysis.NullableAnnotation.NotAnnotated, local.NullableAnnotation);
```

Comment:
>Assert.Equal(CodeAnalysis.NullableAnnotation.NotAnnotated, local.NullableAnnotation); [](start = 12, length = 85)

Consider also verifying inference result for scenarios when interfaces are in different order in AllInterfaces. I would expect the result to not depend on the order.